appdatabase/migrations/sql/1671451982_add_balance_history.up.sql

```diff
@@ -0,0 +1,15 @@
+-- One entry per chain_id, address, currency, block
+-- bitset is used to select time interval granularity. The least significant bit has the coarsest granularity
+-- balance is a serialized big.Int
+CREATE TABLE IF NOT EXISTS balance_history (
+    chain_id UNSIGNED BIGINT NOT NULL,
+    address VARCHAR NOT NULL,
+    currency VARCHAR NOT NULL,
```

Comment:
I use this to identify the currency for which the balance was cached (ISO), which, along with chain id and address, is part of the unique balance identity. This is the way to get the ground truth, and any other value reinterpretation will add errors and complexity that may propagate and generate side effects if we want to do anything other transformations.

Your example will work only if we want to show also the history of the currency converted to USD. That means showing the USD value at the time and not accounting for inflation. We want to display the historical amount with the current conversion (adjusted for inflation).

Thanks for the heads up; I check this point of view with the design.

Reply:
Good point, regarding the difference between various currency. The trade off is that it will cost more request, I would imagine that showing the value according to current currency values is what makes more sense.

But yes better check with John